src/main/java/usi/si/seart/service/StaticCodeAnalysisServiceImpl.java

```diff
@@ -61,7 +57,7 @@ public Future<Set<GitRepoMetric>> getCodeMetrics(@NotNull GitRepo repo, boolean
             metrics = convert(repo, g.fromJson(output, JsonObject.class));
             repo.setMetrics(metrics);
             repo.setCloned();
-            gitRepoRepository.save(repo);
+            gitRepoService.createOrUpdateRepo(repo);
```

Comment:
I think I remember now why we used to use `createOrUpdateRepo`, and I don't think we can use it here. I can explain it in more detail to you personally, but as of right now I think we should use the repository directly, so revert this change.

Reply:
I'll swing by the office so you can explain it later :)